graspologic/plot/plot.py

```diff
@@ -671,6 +674,248 @@ def pairplot(
+    # must give labels AND palettes
+    if (labels is not None) and (cluster_palette is None or label_palette is None):
+        msg = "if give labels array must give palette dicts or string desciptors"
+        raise ValueError(msg)
+    elif labels is None and (cluster_palette is not None or label_palette is not None):
+        msg = "must give labels if using palettes for labels and clusters"
+        raise ValueError(msg)
+    # This takes care of default case, when no palettes or labels given
+    elif labels is None and cluster_palette is None and label_palette is None:
+        colors = sns.color_palette("Set1", n_components)
+        cluster_palette = dict(zip(np.unique(lab_names), colors))
+        label_palette = dict(zip(np.unique(lab_names), colors))
+
+    with sns.plotting_context(context=context, font_scale=font_scale):
+        # handle the case where only given two dims
+        if X.shape[1] == 2:
+            pairplot, axes = plt.subplots(1, 1, figsize=(12, 12))
+            _plot_ellipse(
+                data,
+                X,
+                0,
+                1,
+                means,
+                covariances,
+                axes,
+                label_palette,
+                cluster_palette,
+                alpha=alpha,
+            )
+            # formatting
+            axes.set_ylabel("Dim " + str(0))
+            axes.set_xlabel("Dim " + str(1))
+            axes.label_outer()
+            axes.spines["right"].set_visible(False)
+            axes.spines["top"].set_visible(False)
+            handles, labels = [], []
+            handles_, labels_ = axes.get_legend_handles_labels()
+            handles += handles_
+            labels += labels_
+            pairplot.legend(
+                handles[: means.shape[0] + 1],
+                labels[: means.shape[0] + 1],
+                loc="upper right",
+            )
+            if title:
+                pairplot.suptitle(title)
+            else:
+                pairplot.suptitle(
+                    "Fit a Gaussian mixture using "
+                    + str(means.shape[0])
+                    + " components"
+                )
+            return pairplot
+        # for the case with more than 2 dims
+        pairplot, axes = plt.subplots(X.shape[1], X.shape[1], figsize=(12, 12))
+        for k in range(X.shape[1]):
+            for j in range(X.shape[1]):
+                if k == j:
+                    # take care of the distplot on diagonal
+                    for t, lab in zip([i for i in range(X.shape[1])], label_palette):
+                        sns.distplot(
+                            X[Y_ == t, k],
+                            kde=True,
+                            ax=axes[k, k],
+                            color=label_palette[lab],
+                        )
+                    axes[k, j].set_xticks([]), axes[k, j].set_yticks([])
+                    axes[k, j].set_ylabel(None)
```

Comment:
why is this line necessary?

Reply:
This removes the tick marks on the graph to make it look like the original pairplot output. https://github.com/microsoft/graspologic/pull/541/commits/4f36a49c66cdb2e409f21ed08d6578667c0c00a7 i have it explained in the comments here